doc/build-netbsd.md

```diff
@@ -1,81 +1,145 @@
--------------
+## Dependencies
 
-You will need the following modules, which can be installed via pkgsrc or pkgin:
+The following dependencies are **required**:
+
+ Library                                          | Purpose    | Description
+ -------------------------------------------------|------------|----------------------
+ [autoconf](https://pkgsrc.se/devel/autoconf)     | Build      | Automatically configure software source code
+ [automake](https://pkgsrc.se/devel/automake)     | Build      | Generate makefile (requires autoconf)
+ [libtool](https://pkgsrc.se/devel/libtool)       | Build      | Shared library support
+ [pkg-config](https://pkgsrc.se/devel/pkg-config) | Build      | Configure compiler and linker flags
+ [git](https://pkgsrc.se/devel/git)               | Clone      | Version control system
+ [gmake](https://pkgsrc.se/devel/gmake)           | Compile    | Generate executables
+ [boost](https://pkgsrc.se/meta-pkgs/boost)       | Utility    | Library for threading, data structures, etc
+ [libevent](https://pkgsrc.se/devel/libevent)     | Networking | OS independent asynchronous networking
+
+
+The following dependencies are **optional**:
+
+  Library                                            | Purpose          | Description
+  ---------------------------------------------------|------------------|----------------------
+  [db4](https://pkgsrc.se/databases/db4)             | Berkeley DB      | Wallet storage (only needed when wallet enabled)
+  [qt5](https://pkgsrc.se/x11/qt5)                   | GUI              | GUI toolkit (only needed when GUI enabled)
+  [qrencode](https://pkgsrc.se/converters/qrencode)  | QR codes in GUI  | Generating QR codes (only needed when GUI enabled)
+  [zeromq](https://pkgsrc.se/net/zeromq)             | ZMQ notification | Allows generating ZMQ notifications (requires ZMQ version >= 4.0.0)
+  [sqlite3](https://pkgsrc.se/databases/sqlite3)     | SQLite DB        | Wallet storage (only needed when wallet enabled)
+  [python37](https://pkgsrc.se/lang/python37)        | Testing          | Python Interpreter (only needed when running the test suite)
+
+  See [dependencies.md](dependencies.md) for a complete overview.
+
+
+## Preparation
+
+### 1. Install Required Dependencies
+Install the required dependencies the usual way you [install software on NetBSD](https://www.netbsd.org/docs/guide/en/chap-boot.html#chap-boot-pkgsrc) - either with `pkg_add` or `pkgin`. The example commands below use `pkgin` which is usually run as `root` or via `sudo`.
+
+```bash
+pkgin install autoconf automake libtool pkg-config git gmake boost libevent
 
 ```
-autoconf
-automake
-boost
-git
-gmake
-libevent
-libtool
-pkg-config
-python37
 
+See [dependencies.md](dependencies.md) for a complete overview.
+
+### 2. Clone Bitcoin Repo
+Now that `git` and all the required dependencies are installed, let's clone the Bitcoin Core repository to a directory. All build scripts and commands will run from this directory.
+``` bash
```

Comment:
I am using a completely clean os...

```
netbsd# git clone https://github.com/bitcoin/bitcoin
Cloning into 'bitcoin'...
error: failed to write new configuration file /root/bitcoin/.git/config.lock
warning: unable to unlink '/root/bitcoin/.git/config.lock': Operation not permitted
fatal: could not set 'core.repositoryformatversion' to '0'
netbsd# 
```

setting a better testing environment for NetBSD...

Reply:
That should be an error in the system side. First, make sure you have proper read/write access to that directory from your user. I could be wrong, but it looks like you are trying to write somewhere on root from a user. Some other troubleshooting tips: try allocating more space for the netBSD partition or manually deleting the config.lock which could have been put there by another program or process.

Reply:
Here is a docker repo for testing NetBSD9.#

https://github.com/bitcoincore-dev/core-review/tree/docker-netbsd9

Works - figured others might find it useful...